src/compiler/checker.ts

```diff
@@ -1093,38 +1093,71 @@ namespace ts {
             return links.resolvedExports || (links.resolvedExports = getExportsForModule(moduleSymbol));
         }
 
-        function extendExportSymbols(target: SymbolTable, source: SymbolTable) {
+        interface ExportedSymbolDiagnostics {
+            specifierText: string;
+            exportsWithDuplicate: ExportDeclaration[];
+        }
+
+        function extendExportSymbols(target: SymbolTable, source: SymbolTable, lookupTable?: Map<ExportedSymbolDiagnostics>, exportNode?: ExportDeclaration) {

```

Comment:
At a cursory glance, it's hard for me to understand exactly what's going on. Can you add a quick comment for what this is and what it's used for?

For instance, what does `lookupTable` do? Which of these are you going to modify?